vembrane/modules/filter.py

```diff
@@ -62,46 +62,89 @@ def add_subcommmand(subparsers):
         help="Keep all annotations of a variant if at least one of them passes "
         "the expression.",
     )
+    parser.add_argument(
+        "--preserve-order",
+        default=False,
+        action="store_true",
+        help="Make sure that the variant order of the output matches that of the \
+              input. For performance reasons only make use of this option, if the \
+              input might contain breakends (BNDs) as the order of all other variants \
+              are preserved anyway.",
+    )
+
+
+def test_and_update_record(env, idx, record, ann_key, keep_unmatched):
+    env.update_from_record(idx, record)
+    if env.expression_annotations():
+        # if the expression contains a reference to the ANN field
+        # get all annotations from the record.info field
+        # (or supply an empty ANN value if the record has no ANN field)
+        try:
+            annotations = record.info[ann_key]
+        except KeyError:
+            annotations = [""]
+        #  … and only keep the annotations where the expression evaluates to true
+        filtered_annotations = [
+            annotation for annotation in annotations if env.evaluate(annotation)
+        ]  # TODO: if keep_unmatched, that could be "any(iterator)" instead of list
+
+        if not keep_unmatched and (len(annotations) != len(filtered_annotations)):
+            # update annotations if they have actually been filtered
+            record.info[ann_key] = filtered_annotations
+
+        return record, len(filtered_annotations) > 0
```

Comment:
is `len(filtered_annotation) > 0` the correct "pass" bool when `keep_unmatched is True`?

Reply:
Yes. If all annotations are filtered then the record becomes invalid, regardless of keep_unmatched.
If keep_unmatched is True, the only consequence is, that the record annotations are not updated.